ext/opentelemetry-ext-datadog/src/opentelemetry/ext/datadog/exporter.py

```diff
@@ -128,6 +131,11 @@ def _translate_to_datadog(self, spans):
 
             datadog_span.set_tags(span.attributes)
 
+            # add origin to root span
+            origin = _get_origin(span)
+            if origin and parent_id == 0:
+                datadog_span.set_tag(DD_ORIGIN, origin)
```

Comment:
an aside, but how is the origin key used by DataDog? I wonder if this is something to raise up to the specification SIG, if there's value in another key that is not appropriately propagated.

This also raises an issue for those who want to use open propagation specifications and DataDog at the same time. If one does want to use DataDog in it's full capacity, I presume that will require the setting of the DD_ORIGIN value. For those want to switch to DataDog despite using an existing propagation spec, they may not get the full functionality unless they switch all services to use this propagator.

Thoughts?

Reply:
This is used to link traces to Synthetics tests: https://docs.datadoghq.com/synthetics/apm/#how-are-traces-linked-to-tests

As for using Datadog alongside an open propagation specification, I had been thinking that this be accomplished by adding the Datadog format to the composite propagator. Here's how I did it for a test app:

``` python
    global_httptextformat = propagators.get_global_httptextformat()
    if isinstance(
        global_httptextformat, propagators.composite.CompositeHTTPPropagator
    ) and not any(
        isinstance(p, DatadogFormat) for p in global_httptextformat._propagators
    ):
        propagators.set_global_httptextformat(
            propagators.composite.CompositeHTTPPropagator(
                global_httptextformat._propagators + [DatadogFormat()]
            )
        )
    else:
        propagators.set_global_httptextformat(DatadogFormat())
```

Appending to the global format could be something that's added to the API.

All services would have to have the Datadog propagator added, though that could be part of enabling the Datadog exporter for these services.

@toumorokoshi Any problems you see with that approach?

Reply:
The composite approach makes sense as a transition, I was referring more toward using open propagation exclusively. 

But it sounds like that would work perfectly fine here, the only caveat is the missing DD_ORIGIN propagation, which isn't needed for all the features.